pycbc/inference/burn_in.py

```diff
@@ -28,6 +28,59 @@
 
 import numpy
 
+def ks_test(sampler, fp):
+    """Burn in based on whether the p-value of the KS test between the samples
+    at the last iteration and the samples midway along the chain for each
+    parameter is > 0.1 and < 0.9
+
+    Parameters
+    ----------
+    sampler : pycbc.inference.sampler
+        Sampler to determine burn in for. May be either an instance of a
+        `inference.sampler`, or the class itself.
+    fp : InferenceFile
+        Open inference hdf file containing the samples to load for determing
+        burn in.
+
+    Returns
+    -------
+    burn_in_idx : array
+        Array of indices giving the burn-in index for each chain.
+    is_burned_in : array
+        Array of booleans indicating whether each chain is burned in.
+    """
+    from scipy.stats import ks_2samp
+    from scipy import interpolate
+    nwalkers = fp.nwalkers
+    niterations = fp.niterations
+    burn_in_idx = numpy.repeat((niterations-1), nwalkers).astype(int)
+    # The chains should not be burned in by default. So is_burned_in is assigned
+    # a False value for all the walkers by default.
+    is_burned_in = numpy.zeros(nwalkers, dtype=bool)
+    # Create a dictionary which would have keys are the variable args and values
+    # are booleans indicating whether the p-value for the parameters satisfies
+    # the KS test
+    is_burned_in_param = {}
+    # iterate over the parameters
+    for param in fp.variable_args:
+        # read samples for the parameter from the last iteration of the chain
+        samples_last_iter = sampler.read_samples(fp, param, iteration=-1,
+                                                 flatten=True)[param]
+        # read samples for the parameter from the iteration midway along the chain
+        samples_chain_midpt = sampler.read_samples(fp, param, iteration=int(niterations/2),
+                                                  flatten=True)[param]
+        _, p_value = ks_2samp(samples_last_iter, samples_chain_midpt)
+        # check if p_value is within the desired range
+        if p_value > 0.1 and p_value < 0.9 :
+            is_burned_in_param[param] = True
+        else :
+            is_burned_in_param[param] = False
```

Comment:
Just do:
```
is_burned_in_param[param] = p_value > 0.1 and p_value < 0.9
```

Reply:
BTW, here's a really nice Pythonism I came across the other day
```
is_burned_in_param[param] = 0.1 < p_value < 0.9
```
This does the right thing, in particular `p_value` is only evaluated once
https://docs.python.org/2/reference/expressions.html#grammar-token-comp_operator

Reply:
Thanks @lppekows! That was a really cool Python trick of the day! :D